### PR TITLE
Merge TF and PT CLI

### PR DESCRIPTION
**Pull request by njzjz**
Just merge in form. Several options or subcommands are only supported by TensorFlow or PyTorch.

Also, avoid import from `deepmd.tf` in `deepmd.utils.argcheck`.

```
Use --tf or --pt to choose the backend:
    dp --tf train input.json
    dp --pt train input.json
```

### CODE REVIEW

File: deepmd/main.py [hunk 1]
```diff
@@ -56,8 +57,48 @@ def main_parser() -> argparse.ArgumentParser:
...
+        raise ValueError(
+            f"Unknown backend {default_backend}. "
+            "Please set DP_BACKEND to either tensorflow or pytorch."
+        )
+
+    parser_backend = parser.add_mutually_exclusive_group()
+    parser_backend.add_argument(
+        "-b",
+        "--backend",
+        choices=["tensorflow", "pytorch"],
```
Comment (wanghan-iapcm) [inline deepmd/main.py:82]: shall we provide short-hand keyword like `tf` and `pt` ?

File: deepmd/main.py [hunk 2]
```diff
@@ -651,6 +733,16 @@ def main():
         if no command was input
     """
     args = parse_args()
-    from deepmd.tf.entrypoints.main import main as deepmd_main
+
+    if args.tf:
```
Comment (wanghan-iapcm) [inline deepmd/main.py:737]: we may want to check `--tf`` and `--pt` should be be set at the same time.

Comment (njzjz) [reply deepmd/main.py:737]: This is what `add_mutually_exclusive_group` has done

Comment (njzjz) [reply deepmd/main.py:737]: It seems there is a better code here:
https://stackoverflow.com/a/62863301/9567349

File: deepmd/main.py (hunk 1 above)
Comment (njzjz) [reply deepmd/main.py:82]: I found an elegant way here: https://stackoverflow.com/a/72651934/9567349

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**